app/promptfoo/readability_assessment.py

```diff
@@ -0,0 +1,108 @@
+from typing import Dict, Union, Any
+import textdescriptives as td
+import numpy as np
+
+def get_assert(output: str, context) -> Union[bool, float, Dict[str, Any]]:
+    """
+    Assess the readability of the output text using TextDescriptives instead of py-readability-metrics.
+    Returns a GradingResult with component scores for different readability metrics.
+    """
+    print("=== TEXTDESCRIPTIVES READABILITY ASSESSMENT STARTING ===")
+    print(f"Output to assess: {output}")
+    
+    try:
+        if not output or len(output.strip()) == 0:
+            return {
+                'pass': False,
+                'score': 0.0,
+                'reason': 'Empty or invalid output text'
+            }
+        
+        # Use TextDescriptives to calculate readability metrics
+        metrics_df = td.extract_metrics(
+            text=output, 
+            spacy_model="en_core_web_sm", 
+            metrics=["readability"]
+        )
+        
+        # Extract the readability metrics and convert from numpy types to Python native types
+        flesch_reading_ease = float(metrics_df["flesch_reading_ease"].iloc[0])
+        flesch_kincaid_grade = float(metrics_df["flesch_kincaid_grade"].iloc[0])
+        gunning_fog = float(metrics_df["gunning_fog"].iloc[0])
+        coleman_liau_index = float(metrics_df["coleman_liau_index"].iloc[0])
```

Comment:
Refactor into a separate helper method that returns a dict with keys "flesch_reading_ease", ... and float values. The dict can be set as the value for `namedScores` in line 87.

Reply:
I'd also move line 39-40:
```
grade_levels = [flesch_kincaid_grade, gunning_fog, coleman_liau_index]
avg_grade_level = sum(grade_levels) / len(grade_levels)
```
into that helper function.